lib/http.js

```diff
@@ -284,8 +284,15 @@ module.exports = function (chai, _) {
     new Assertion(this._obj).to.redirect;
 
     if(redirects && redirects.length) {
+      var hasRedirected;
+
+      if (Object.prototype.toString.call(destination) === '[object RegExp]') {
+        hasRedirected = redirects.filter(redirect => destination.test(redirect)).length > 0;
```

Comment:
I like the use of ES5 array methods! You might also consider using a [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) so that we don't have to test more redirects if we already know one of the previous redirects matched.

```js
hasRedirected = redirects.reduce((hasMatched, redirect) => hasMatched || destination.test(redirect), false);
```

Reply:
Oh nice, yeah that is a little more appropriate.